parsers/regex/regex_test.go

```diff
@@ -0,0 +1,168 @@
+package regex
+
+import (
+	"errors"
+	"reflect"
+	"regexp"
+	"testing"
+	"time"
+
+	"github.com/honeycombio/honeytail/event"
+	"github.com/honeycombio/honeytail/parsers"
+)
+
+const (
+	commonLogFormatTimeLayout = "02/Jan/2006:15:04:05 -0700"
+	iso8601TimeLayout         = "2006-01-02T15:04:05-07:00"
+)
+
+type testLineMap struct {
+	lineRegex string
+	input     string
+	expected  map[string]interface{}
+}
+
+// Test cases for RegexLineParser.ParseLine
+var tlms = []testLineMap{
+	{
+		// Simple word parsing
+		lineRegex: `(?P<word1>\w+) (?P<word2>\w+) (?P<word3>\w+)`,
+		input:     `apple banana orange`,
+		expected: map[string]interface{}{
+			"word1": "apple",
+			"word2": "banana",
+			"word3": "orange",
+		},
+	},
+	{
+		// Simple time parsing
+		lineRegex: `(?P<Year>\d{4})-(?P<Month>\d{2})-(?P<Day>\d{2})`,
+		input:     `2017-01-30 1980-01-02`, // Ignore the second date
+		expected: map[string]interface{}{
+			"Year":  "2017",
+			"Month": "01",
+			"Day":   "30",
+		},
+	},
+	{
+		// Fields containing whitespace
+		lineRegex: `\[(?P<BracketedField>[0-9A-Za-z\s]+)\] (?P<UnbracketedField>[0-9A-Za-z]+)`,
+		input:     `[some value] unbracketed`,
+		expected: map[string]interface{}{
+			"BracketedField":   "some value",
+			"UnbracketedField": "unbracketed",
+		},
+	},
+	{
+		// Nested regex grouping
+		lineRegex: `(?P<outer>[^ ]* (?P<inner1>[^ ]*) (?P<inner2>[^ ]*))`,
+		input:     `foo bar baz`,
+		expected: map[string]interface{}{
+			"outer":  "foo bar baz",
+			"inner1": "bar",
+			"inner2": "baz",
+		},
+	},
+	{
+		// Sample nginx log line
```

Comment:
nit: please clarify to `nginx error log line` to differentiate it from the `access.log` line that the nginx parser usually gets.